src/Scripting/Core/ScriptOptions.cs

```diff
@@ -282,5 +295,17 @@ private static MetadataReference CreateReferenceFromAssembly(Assembly assembly)
         /// <exception cref="ArgumentNullException"><paramref name="imports"/> is null or contains a null reference.</exception>
         public ScriptOptions AddImports(params string[] imports) =>
             AddImports((IEnumerable<string>)imports);
+
+        /// <summary>
+        /// Creates a new <see cref="ScriptOptions"/> with debugging information enabled.
+        /// </summary>
+        public ScriptOptions WithDebugInformation() =>
+            new ScriptOptions(this) { EmitDebugInformation = true };
+
+        /// <summary>
+        /// Creates a new <see cref="ScriptOptions"/> with specified <see cref="FileEncoding"/>.
+        /// </summary>
+        public ScriptOptions WithFileEncoding(Encoding encoding) =>
+            new ScriptOptions(this) { FileEncoding = encoding };
```

Comment:
>new ScriptOptions(this) [](start = 12, length = 23)

Check if FileEncoding == encoding to avoid creating new instance if the values are equal.